### ISSUE TITLE
Add top CLI Commands in `CLI Overview`

### ISSUE THREAD

**Issue by paolaperaza** [astronomer/docs#2157]
### Context

We recently published a great Astro quickstart that takes users through installing the CLI, creating a project, creating a Deployment, and pushing code to that Deployment. This is great for any Astro customer getting started.

With that said, I recently got feedback from @virajmparekh that there's an opportunity to have some more getting started content in our Astro CLI tab. Specifically, there's no obvious way for a user to understand what the top commands are or what the CLI can do beyond the few sentences we write to describe it.

### Ask

Instead of writing another quickstart that would conflict or be redundant with our existing Astro quickstart, I think we should:

- Consider a new `Top Commands` or `Popular Commands` or `Main Commands` section here: https://docs.astronomer.io/astro/cli/overview
- Add the following commands:
     - `astro dev init`
     - `astro dev start`
     - `astro dev restart`
     - `astro deploy`
- We should have just brief explanations of these commands and a link to their corresponding entry in the command reference doc

(PS. I'd also like to see us move away from the "boxes" currently in the CLI overview and just go back to written form of these sections -- but we can discuss)

**Comment by virajmparekh** [comment]
Thanks Paola! Just to be explicit about what I think the need here is, here's a few user stories (motivated by the last few workshops/meetups):

As a user:

- I want to use the AstroCLI purely for the local development experience it offers me. I don't think I'm ready for Astro as a paid for product, but the CLI offers me real value. As such, when I go to the CLI page, I just want to see how to get to the point where I have an Airflow environment

- Once I've figured out the commands for getting an Airflow env up, I'm running into issues around ports already being allocated. It'd be great to know how to debug a couple of common issues that come up when getting started, be told explicitly how to log in, etc. etc. 

- Lastly, I might need to customize my environment based on a `docker-compose -override.yaml` When I searched this on the Astro docs, it brings me to either software page or [this page on Astro](https://docs.astronomer.io/astro/test-and-troubleshoot-locally#override-the-cli-docker-compose-file). The information is what I need, but it feels a bit weird to be moving out of the CLI tab. 

So, TL;DR is that there's a lot of CLI content that you need to navigate away from the CLI tab from to navigate. If you're just trying to use the CLI as it is, this makes it a bit hard to figure out what you're supposed to do. Totally makes sense if this is a journey we just don't want to optimize for, but just want to make sure that's the case.
